tests/manage/rgw/test_host_node_failure.py

```diff
@@ -98,17 +99,24 @@ def test_rgw_host_node_failure(
                     selector=constants.RGW_APP_LABEL,
                 )
 
-                # Create OBC and read wnd write
-                self.create_obc_creation(bucket_factory, mcg_obj, "Object-key-1")
-
                 # Start the node
                 nodes.start_nodes(node_obj)
 
+                # Check the ceph health OK
+                ceph_health_check(tries=90, delay=15)
+
+                # Verify all storage pods are running
+                wait_for_storage_pods()
+
                 # Create OBC and read wnd write
                 self.create_obc_creation(bucket_factory, mcg_obj, "Object-key-2")
 
+            else:
+                pytest.skip(
+                    "RGW pod and NooBaa DB are not hosted on same node. "
+                    f"RGW pod hosted on node: {pod_node} "
+                    f"NooBaa DB pod hosted on node: {noobaa_pod_node.name}"
+                )
```

Comment:
You're looping through RGW pods here and there could be a situation where you've got 2 RGW pods running on different node than the Noobaa pod.
In this case, you'll skip the test after one iteration and might leave some leftovers (OBC).
I'll give example to make it more understandable:
2 RGW pods: pod_1 runs on node A, pod_2 runs on node B.
Noobaa pod running on node A.

Here the 1st iteration will go well and create the OBC.
The 2nd iteration will skip the test (enter the "else" block) and leave the OBC (created during the previous iteration)

Reply:
Thank you Yosi, for pointing it out. Now validating before the test executed